src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Completion/DirectiveAttributeCompletionItemProvider.cs

```diff
@@ -140,15 +154,42 @@ internal static ImmutableArray<RazorCompletionItem> GetAttributeCompletions(
 
             var razorCompletionItem = RazorCompletionItem.CreateDirectiveAttribute(
                 displayText,
-                insertText,
+                insertTextSpan.ToString(),
```

Comment:
>insertTextSpan.ToString(), [](http://example.com/codeflow?start=16&length=26)

In the common case, aren't we allocating something for insertTextSpan whereas we didn't before?

Reply:
> I don't think we actually allocate here

Isn't that specialization you linked to doing a string allocation? Either way, the check you added makes that no longer a concern.

```
        public override string ToString()
        {
            if (typeof(T) == typeof(char))
            {
                return new string(new ReadOnlySpan<char>(ref Unsafe.As<T, char>(ref _reference), _length));
            }
            return $"System.ReadOnlySpan<{typeof(T).Name}>[{_length}]";
        }
```

Reply:
It's a similar one, but I did say later ".. or maybe not, looking at it more closely" :)